src/sdk/actions.ts

```diff
@@ -130,3 +142,47 @@ async function returnDataToKernel(repoToken: string, stateId: string, output: ob
     },
   });
 }
+
+function createStructuredMetadata(className: string | undefined, logReturn: LogReturn) {
+  let logMessage, metadata;
+  if (logReturn) {
+    logMessage = logReturn.logMessage;
+    metadata = logReturn.metadata;
+  }
+
+  const jsonPretty = sanitizeMetadata(metadata);
+  const stackLine = new Error().stack?.split("\n")[2] ?? "";
+  const caller = stackLine.match(/at (\S+)/)?.[1] ?? "";
+  const ubiquityMetadataHeader = `<!-- ${HEADER_NAME} - ${className} - ${caller} - ${metadata?.revision}`;
```

Comment:
We should maybe place `\n\n` before the beginning of the comment so it's detached from any other text and formatting isn't at risk.

Reply:
I wrapped the final results with carriage returns for safety.